bundles/specmate-persistency-api/src/com/specmate/persistency/ITransaction.java

```diff
@@ -1,43 +1,45 @@
 package com.specmate.persistency;
 
+import com.specmate.common.RestResult;
 import com.specmate.common.SpecmateException;
 import com.specmate.common.SpecmateValidationException;
 
 /**
  * Encapsulates the access to a model repository.
- * 
+ *
  * @author junkerm
  *
  */
 public interface ITransaction extends IView {
 
 	/**
 	 * Commits changes done in this transaction to the repository
-	 * 
+	 *
 	 * @throws SpecmateException
-	 *             If the commit could not be perfomed successfully
+	 *             If the commit could not be performed successfully
 	 */
 	public void commit() throws SpecmateException;
 
+	public void commit(String userName) throws SpecmateException;
+
 	/** Rolls back changes made in this transaction since the last commit */
 	public void rollback();
 
-	/** Closes the transcation. */
+	/** Closes the transaction. */
 	public void close();
 
-	/** Signals if uncomitted changes exist in the transaction */
+	/** Signals if uncommitted changes exist in the transaction */
 	boolean isDirty();
 
 	/**
 	 * Perform a change and commit
-	 * 
+	 *
 	 * @throws SpecmateValidationException
 	 */
-	<T> T doAndCommit(IChange<T> change) throws SpecmateException, SpecmateValidationException;
+	<T> RestResult<T> doAndCommit(IChange<T> change) throws SpecmateException, SpecmateValidationException;

```

Comment:
The transaction should be indedependent of the used interface. I would not use RestResult as a return type

Reply:
Ok, reverted at the cost of manual type casting.